app/models/player.rb

```diff
@@ -57,11 +59,19 @@ def had_bye?
   end
 
   def corp_identity_object
-    Identity.find_or_initialize_by(name: corp_identity)
+    if corp_identity_ref
+      corp_identity_ref
+    else
+      Identity.find_or_initialize_by(name: corp_identity)
```

Comment:
is this going to force identities where they were effectively optional before?

Reply:
No, it's the same code that was there before, but with an optional lookup by ID if the field has been set. It'll fall back to the behaviour that was there before if nothing was entered, if the entered ID doesn't exist, or if at the time it was saved to the player record, the identities were out of date and it hadn't been loaded into Cobra yet.